### PR TITLE
Form post request

### PR DESCRIPTION
**Pull request by mam249**
wasn't sure if this is how you're meant to resolve nits but I just added it in as an extra commit here

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -28,24 +28,45 @@
...
+    String displayName = getParameter(request, "display-name", "");
+    boolean anonymous = Boolean.parseBoolean(getParameter(request, "anonymous", "false"));
+
+    if (anonymous) {
+        displayName = "Anonymous";
+    }
+
+    if (displayName == ""){
+        displayName = "Anonymous";
+    }
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]:
```suggestion
    if (anonymous || displayName == "") {
        displayName = "Anonymous";
    }
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -28,24 +28,45 @@
...
+      comments_array = new ArrayList<>();
+  }  
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Get the input from the form.
+    String comment = getParameter(request, "comment-box", "");
+    String displayName = getParameter(request, "display-name", "");
+    boolean anonymous = Boolean.parseBoolean(getParameter(request, "anonymous", "false"));
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: false (no quotes)

**Review by lminchillo — CHANGES_REQUESTED** [review]
LGTM after replacing "false" with false

**Comment by mam249** [comment]
when I replace "false" with false the getParameter function stops working - I took the function as is from one of the examples

**Comment by lminchillo** [comment]
> when I replace "false" with false the getParameter function stops working - I took the function as is from one of the examples

Oh ok I got it now. It's because parseBoolean accepts a String so that's actually correct. My bad!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -28,24 +28,40 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
-  private ArrayList<String> trial_array;
+public final class DataServlet extends HttpServlet {
+  private ArrayList<String> comments_array;
   private String json_array;
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: Please change to camel case

https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names

**Review by lminchillo — APPROVED** [review]
LGTM!

**Pull request merged**